src/coreclr/jit/emitxarch.cpp

```diff
@@ -11789,17 +11789,8 @@ const char* emitter::emitRegName(regNumber reg, emitAttr attr, bool varName) con
 
         case EA_4BYTE:
         {
-            if (IsXMMReg(reg))
-            {
-                return emitXMMregName(reg);
-            }
-
+            assert(isGeneralRegister(reg));
 #if defined(TARGET_AMD64)
-            if (reg > REG_R31)
-            {
-                break;
-            }
```

Comment:
This one is fine to remove and can be replaced with the `assert(isGeneralRegister(reg));`